npm v7 incompatibility

### Version
4.5.7

### Reproduction link
[https://github.com/rajeshjaga/vue-cli-report](https://github.com/rajeshjaga/vue-cli-report)





### Environment info
```
  System:
    OS: Windows 10 10.0.19042
    CPU: (4) x64 Intel(R) Core(TM) i3-6006U CPU @ 2.00GHz
  Binaries:
    Node: 14.11.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.0.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 85.0.4183.121
    Edge: Spartan (44.19041.423.0), Chromium (86.0.622.38)
  npmGlobalPackages:
    @vue/cli: Not Found

```


### Steps to reproduce
clone the linked repo 
npm install
&& 
to resolve this prob:
change eslint version in package.json 7.11.0

### What is expected?
vue create app
created successfully

### What is actually happening?
 code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: eslint@7.11.0
npm ERR! node_modules/eslint
npm ERR! dev eslint@"^6.7.2" from the root project



---
Please bump up eslint version to 7.11.0 
I resolve it by changing the version on package.json

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/@vue/cli/lib/util/ProjectPackageManager.js
+++ b/packages/@vue/cli/lib/util/ProjectPackageManager.js
@@ -134,6 +134,10 @@ class PackageManager {
 
         this.needsNpmInstallFix = true
       }
+
+      if (semver.gte(npmVersion, '7.0.0')) {
+        this.needsPeerDepsFix = true
+      }
     }
 
     if (!SUPPORTED_PACKAGE_MANAGERS.includes(this.bin)) {
@@ -374,7 +378,7 @@ class PackageManager {
       return
     }
 
-    return await this.runCommand('install')
+    return await this.runCommand('install', this.needsPeerDepsFix ? ['--legacy-peer-deps'] : [])
   }
 
   async add (packageName, {
@@ -388,6 +392,10 @@ class PackageManager {
       } else {
         process.env.npm_config_save_prefix = '~'
       }
+    }
+
+    if (this.needsPeerDepsFix) {
+      args.push('--legacy-peer-deps')
     }
 
     return await this.runCommand('add', [packageName, ...args])

